libhwmon/wrapadl.cpp

```diff
@@ -188,6 +206,18 @@ int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanp
 	return 0;
 }
 
+int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int* miliwatts)
+{
+	wrap_adlReturn_t rc;
+	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount){
+		return -1;
+	}
+	int power = 0;
+	rc = adlh->adl2Overdrive6CurrentPowerGet(adlh->context, adlh->phys_logi_device_id[gpuindex], 0, &power);
+	*miliwatts = power * 3.90625;
```

Comment:
@Davesmacer: may I ask where `3.90625` comes from?

Reply:
@jean-m-cyr  might answer that better... Is a conversion from the overdrive format (8 fractional bits)

Reply:
I was looking into the C4244 MSVC warnings and I was wondering if this magic number was something to round things.

What do you think about using `round` and also casting to `unsigned`?

Reply:
Why? I like having decimals, even if I don't use them in the program, i use them in the server that uses getminerstathr. In 10+ cards, 0.1 difference becomes 1 complete Watt...

Reply:
I thought the end result was still unsigned without any decimals? Maybe I missed something.

Reply:
You're totally right, I forgot about the method... it returns miliwatts unsigned.

Reply:
But what does a round and a cast that is not doing already?

Reply:
Well, at least, we wouldn't be asking this :) Because it would be clear what the intention was.

Reply:
Yeah right! I guess that would be nice to add in a future formatting PR, alongside with some comments in the keys of the code...

Reply:
I feel like we shouldn't silence any compiler warnings BTW, because this is what made me look into it.

Anyway, thanks for the info, let's not get too off topic here :)